processing/src/main/java/io/druid/query/groupby/GroupByQuery.java

```diff
@@ -281,10 +291,100 @@ public int compare(Object lhs, Object rhs)
     );
   }
 
-  public Ordering<Row> getRowOrdering(final boolean granular)
+  public Ordering<Row> getRowOrderingForPushDown(
+      final boolean granular,
+      final DefaultLimitSpec limitSpec,
+      final List<AggregatorFactory> aggregatorSpecs
+  )
   {
     final boolean sortByDimsFirst = getContextSortByDimsFirst();
 
+    final List<String> orderedFieldNames = new ArrayList<>();
+    final Set<Integer> dimsInOrderBy = new HashSet<>();
+    final List<Integer> directions = new ArrayList<>();
+
+    for (OrderByColumnSpec orderSpec : limitSpec.getColumns()) {
+      int direction = orderSpec.getDirection() == OrderByColumnSpec.Direction.ASCENDING ? 1 : -1;
+      int dimIndex = OrderByColumnSpec.getDimIndexForOrderBy(orderSpec, dimensions);
+      if (dimIndex >= 0) {
+        DimensionSpec dim = dimensions.get(dimIndex);
+        orderedFieldNames.add(dim.getOutputName());
+        dimsInOrderBy.add(dimIndex);
+        directions.add(direction);
+      } else {
+        int aggIndex = OrderByColumnSpec.getAggIndexForOrderBy(orderSpec, aggregatorSpecs);
+        if (aggIndex >= 0) {
+          AggregatorFactory agg = aggregatorSpecs.get(aggIndex);
+          orderedFieldNames.add(agg.getName());
+          directions.add(direction);
+        }
```

Comment:
getRowOrderingPushDown(..) is only used when limitSpec does not have aggregator in it, so shoudnt' this case of aggIndex >= 0 should actually be an error